### ISSUE TITLE
Cloud trace monitoring crashed and stopped working

### ISSUE THREAD

**Issue by baole** [spring-cloud/spring-cloud-gcp#2243]
**Describe the bug**

I integrated `spring-cloud-gcp-starter-trace` in our project for Google Cloud App Engine Flexible/Java. It started working nicely. However, after sometime, it crashed and stopped working. 

Here is the crash stacktrace:
```
java.lang.IllegalStateException: timeout waiting for onClose. timeoutMs=5000, resultSet=false
	at zipkin2.reporter.stackdriver.internal.AwaitableUnaryClientCallListener.await(AwaitableUnaryClientCallListener.java:49) ~[zipkin-sender-stackdriver-0.15.2.jar!/:na]
	at zipkin2.reporter.stackdriver.internal.UnaryClientCall.doExecute(UnaryClientCall.java:50) ~[zipkin-sender-stackdriver-0.15.2.jar!/:na]
	at zipkin2.Call$Base.execute(Call.java:380) ~[zipkin-2.19.0.jar!/:na]
	at zipkin2.Call$Mapping.doExecute(Call.java:237) ~[zipkin-2.19.0.jar!/:na]
	at zipkin2.Call$Base.execute(Call.java:380) ~[zipkin-2.19.0.jar!/:na]
	at zipkin2.reporter.AsyncReporter$BoundedAsyncReporter.flush(AsyncReporter.java:285) ~[zipkin-reporter-2.11.0.jar!/:na]
	at zipkin2.reporter.AsyncReporter$Flusher.run(AsyncReporter.java:354) ~[zipkin-reporter-2.11.0.jar!/:na]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_222]
```

And Trace list on Google Cloud console.

![image](https://user-images.githubusercontent.com/1460168/76086478-6f0b8780-5fb4-11ea-818f-cf09c30515c7.png)

Do you have any idea?

**Comment by meltsufin** [comment]
Which version of Spring Cloud GCP are you using?
Do you have any indication that the application as a whole continued running?

cc/ @adriancole

**Comment by baole** [comment]
Spring Cloud version: 1.2.1.RELEASE 
Spring Boot version: 2.2.5.RELEASE

Yes, the application is still running and handling incoming requests,  just no new data is sent to Google Cloud Trace console

**Comment by meltsufin** [comment]
You may try `1.2.2.RELEASE`. Can you reproduce the issue consistently?

**Comment by baole** [comment]
I just deployed `1.2.2.RELEASE`. I will keep you posted.

**Comment by elefeint** [comment]
It should work. Your issue is likely the same as openzipkin/zipkin-gcp#127, which was fixed and pulled in to `1.2.2.RELEASE`.

**Comment by baole** [comment]
There are still some exception logs, however, Cloud Trace is working pretty well.
